sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -48,7 +48,8 @@ private object MsSqlServerDialect extends JdbcDialect {
   }
 
   override def getJDBCType(dt: DataType): Option[JdbcType] = dt match {
-    case TimestampType => Some(JdbcType("DATETIME", java.sql.Types.TIMESTAMP))
+    case TimestampType => if(oldDateTime) {Some(JdbcType("DATETIME", java.sql.Types.TIMESTAMP))} 
+      else {Some(JdbcType("DATETIME2", java.sql.Types.TIMESTAMP))}
```

Comment:
I think you should also add an integration test, see `MySQLIntegrationSuite`.

Reply:
I think this needs some minor style changes too - put the if-else on new lines, spaces around parens and braces.